stl_text/datamodule/doc_classification.py

```diff
@@ -27,17 +49,27 @@ def __init__(self, data_path: str = 'glue_sst2_tiny', vocab_path: Optional[str]
         self.text_transform = None
         self.label_transform = None
         self.datasets = {}
+        self.unk_token = '<unk>'
+        self.tokenizer_type = tokenizer_type
 
     def setup(self, stage):
-        # torchtext's spm_tokenizer and vocab are not pickleable and will fail dataset.map()
-        # tokenizer = spm_tokenizer(download_from_url(PRETRAINED_SP_MODEL["text_unigram_25000"]))
-        # vocab = build_fairseq_vocab("vocab_tiny.txt")
-
-        # sequential_transforms is not torchscriptable
-        # self.text_transform = sequential_transforms(tokenizer, vocab)
+        if self.tokenizer_type=='sentencepiece':
+            if self.vocab_path:
+                tokenizer = sentencepiece_tokenizer(download_from_url(PRETRAINED_SP_MODEL["text_unigram_25000"])).to_ivalue()
+                path_manager = PathManager()
+                with path_manager.open(self.vocab_path, "r",encoding='utf-8') as f:
+                    vocab= build_vocab_from_text_file(f,torch.jit.script(tokenizer))
+
+                vocab = vocab.to_ivalue() #TODO Remove to_ivalue() PR: https://github.com/pytorch/text/pull/1080
+                self.text_transform = TextSequentialTransforms(OrderedDict([('tokenizer',tokenizer),('vocabulary',vocab)]))
+            else:
+                self.text_transform = sentencepiece_processor(download_from_url(PRETRAINED_SP_MODEL["text_unigram_25000"])).to_ivalue() 
```

Comment:
Same here. The workflow should still work even if you use the pybind version (a.k.a. without to_ivalue).